tests/unit/service_test.py

```diff
@@ -257,11 +271,17 @@ def test_parse_repository_tag(self):
         self.assertEqual(parse_repository_tag("url:5000/repo"), ("url:5000/repo", ""))
         self.assertEqual(parse_repository_tag("url:5000/repo:tag"), ("url:5000/repo", "tag"))
 
-    def test_latest_is_used_when_tag_is_not_specified(self):
+    @mock.patch('compose.service.Container', autospec=True)
+    def test_create_container_latest_is_used_when_no_tag_specified(self, mock_container):
+        mock_container.create.side_effect = APIError("oops", mock.Mock())
         service = Service('foo', client=self.mock_client, image='someimage')
-        Container.create = mock.Mock()
-        service.create_container()
-        self.assertEqual(Container.create.call_args[1]['image'], 'someimage:latest')
+        with self.assertRaises(APIError):
+            service.create_container()
+            self.mock_client.pull.assert_called_once_with(
+                'someimage',
+                tag='latest',
+                insecure_registry=False,
+                stream=True)
```

Comment:
Is the `assert_called_once_with` inside this `with` block ever reached, if `create_container` raises an `APIError`?